SYCL/Matrix/element_wise_all_ops_half.cpp

```diff
@@ -0,0 +1,260 @@
+
+       joint_matrix_fill(sg, sub_a, 5.0);
+
+       auto wi_slice_a = sub_a.get_wi_data();
+       for (int i = 0; i < wi_slice_a.length(); i++) {
+         wi_slice_a[i] = wi_slice_a[i] * 3.0;
+       }
+       joint_matrix_store(sg, sub_a,
+                          accA.get_pointer() + (sg_startx * TM) * N +
+                              sg_starty / SG_SZ * TN,
+                          N, matrix_layout::row_major);
+     }); // parallel for
+   }).wait();
+  assert_ops_ref<T, M, N>(bufA.get_access<access::mode::read>(), ref);
+}
+
+template <typename T, size_t M, size_t N>
+void matrix_verify_div(queue q, big_matrix<T, M, N> &A, nd_range<2> &r,
+                       const float ref) {
+  buffer<half, 2> bufA(A.get_data(), range<2>(M, N));
+
+  q.submit([&](handler &cgh) {
+     auto accA = bufA.get_access<access::mode::read_write>(cgh);
+
+     cgh.parallel_for<class imatrix>(r, [accA](nd_item<2> spmd_item) {
+       const auto global_idx = spmd_item.get_global_id(0);
+       const auto global_idy = spmd_item.get_global_id(1);
+       const auto sg_startx = global_idx - spmd_item.get_local_id(0);
+       const auto sg_starty = global_idy - spmd_item.get_local_id(1);
+
+       ext::oneapi::sub_group sg = spmd_item.get_sub_group();
+       joint_matrix<T, TM, TK> sub_a(sg);
+
+       joint_matrix_fill(sg, sub_a, 4.0);
+
+       auto wi_slice_a = sub_a.get_wi_data();
+       for (int i = 0; i < wi_slice_a.length(); i++) {
+         wi_slice_a[i] = wi_slice_a[i] / 2.0;
+       }
+       joint_matrix_store(sg, sub_a,
+                          accA.get_pointer() + (sg_startx * TM) * N +
+                              sg_starty / SG_SZ * TN,
+                          N, matrix_layout::row_major);
+     }); // parallel for
+   }).wait();
+  assert_ops_ref<T, M, N>(bufA.get_access<access::mode::read>(), ref);
+}
+
+template <typename T, size_t M, size_t N>
+void matrix_verify_logic(queue q, big_matrix<T, M, N> &A, nd_range<2> &r,
+                         const float ref) {
+  buffer<half, 2> bufA(A.get_data(), range<2>(M, N));
+
+  q.submit([&](handler &cgh) {
+     auto accA = bufA.get_access<access::mode::read_write>(cgh);
+
+     cgh.parallel_for<class imatrix>(r, [accA](nd_item<2> spmd_item) {
+       const auto global_idx = spmd_item.get_global_id(0);
+       const auto global_idy = spmd_item.get_global_id(1);
+       const auto sg_startx = global_idx - spmd_item.get_local_id(0);
+       const auto sg_starty = global_idy - spmd_item.get_local_id(1);
+
+       ext::oneapi::sub_group sg = spmd_item.get_sub_group();
+       joint_matrix<T, TM, TK> sub_a(sg);
+
+       joint_matrix_fill(sg, sub_a, 5.0);
+
+       auto wi_slice_a = sub_a.get_wi_data();
+       for (int i = 0; i < wi_slice_a.length(); i++) {
+         if (wi_slice_a[i]) {
+           if (wi_slice_a[i] > 2.0 || wi_slice_a[i] >= 2.0 ||
+               wi_slice_a[i] < 2.0 || wi_slice_a[i] <= 2.0) {
+             T val = (wi_slice_a[i] != 2.0) ? wi_slice_a[i] : 2.0;
```

Comment:
I think we should change the testcase according to the rules of "?: " in I think we should change the testcase according to the rules of "?: "in this website: use static_cast for ternary operator in our code.

> If the types of the second and third operands are not identical, then complex type conversion rules, as specified in the C++ Standard, are invoked. These conversions may lead to unexpected behavior including construction and destruction of temporary objects. For this reason, we strongly advise you to either (1) avoid using user-defined types as operands with the conditional operator or (2) if you do use user-defined types, then explicitly cast each operand to a common type.